pan3d/dataset_builder.py

```diff
@@ -261,77 +281,66 @@ def on_set_array_active(self, array_active, **kwargs):
                     "step": 1,
                 }
             )
-            self.state.expanded_coordinates.append(key),
+            self.state.ui_expanded_coordinates.append(key),
         self.auto_select_coordinates()
-        self.state.dirty("coordinates", "expanded_coordinates")
-
-    @change("x_array")
-    def on_set_x_array(self, x_array, **kwargs):
-        self.algorithm.x = x_array
-        self.mesh_changed()
-
-    @change("y_array")
-    def on_set_y_array(self, y_array, **kwargs):
-        self.algorithm.y = y_array
-        self.mesh_changed()
+        self.state.dirty("da_coordinates", "ui_expanded_coordinates")
+        self.plotter.clear()
+        self.plotter.view_isometric()
 
-    @change("z_array")
-    def on_set_z_array(self, z_array, **kwargs):
-        self.algorithm.z = z_array
+    @change("da_x", "da_y", "da_z")
+    def on_set_da_axis(self, da_x, da_y, da_z, **kwargs):
```

Comment:
update_mesh_coordinates?

Reply:
Side note here. Are those methods aiming to be used by the user or internal?

Reply:
If internal then prefix them with an _. 
If you want the user to use them as well, then make sure they can be understood. Like `set_active_array` or `set_axis_array_names`...

Reply:
We could do something similar to what we did for `set_dataset_path`, where the incoming parameter may either be already stored in the state or may be a new change. Like your suggestion for `set_dataset_path`, where we store `self.dataset_path` separately from `self.state.dataset_path` so that we can make the comparison and do a no-op in the case of a match.

Reply:
So should the method name be `set_dataset_axis_names`?

Reply:
Sure, I like that naming. I can apply that pattern to all the state var side-effect functions so that any of them can be either called by the user directly or as a response to a state change.